### DISCUSSION TITLE
ISSUES: can not get new refresh token (valid only 1 hour) & during new setup on step 2 got the following error

### DISCUSSION THREAD

**Discussion by zzack1** [potmat/homebridge-google-nest-sdm#193] [Q&A]
**Have you read the [FAQ](https://github.com/potmat/homebridge-google-nest-sdm#faq)?**

Yes

**Have you searched [existing issues](https://github.com/potmat/homebridge-google-nest-sdm/issues?q=is%3Aissue)?**

Yes

**Have you searched [existing discussions](https://github.com/potmat/homebridge-google-nest-sdm/discussions?discussions_q=)?**

Yes

**Describe the bug**
can not get new refresh token (valid only 1 hour) & during new setup on step 2 got the following error

**Attach Logs**
not yet in hb

**Screenshots**
![Screenshot 2024-11-17 at 2 27 27 PM](https://github.com/user-attachments/assets/b51e59ac-3b88-482a-9ce3-d2c2298085fe)

**Environment**
again not yet in hb, it happens in setup at https://developers.google.com/nest/device-access/authorize

**Additional context**
lets work on this together

**Comment by potmat** [comment]
Converting to discussion as this is not an issue with the plugin itself.

**Comment by travispamaral** [comment]
I was able to solve this by adding `www.google.com` to my apps redirect_uri's in the auth credentials. The url in the google docs for step 2 include this redirect uri so if it is not included in your project credential settings it will error out

<img width="513" alt="image" src="https://github.com/user-attachments/assets/3089fbf2-3f2f-40d5-8668-489468fad811" />
